lib/srv/alpnproxy/auth/auth_proxy.go

```diff
@@ -107,30 +98,20 @@ func (s *AuthProxyDialerService) dialAuthServer(ctx context.Context, clusterName
 }
 
 func (s *AuthProxyDialerService) dialLocalAuthServer(ctx context.Context) (net.Conn, error) {
-	authServers, err := s.accessPoint.GetAuthServers()
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	if len(authServers) == 0 {
+	if len(s.authServers) == 0 {
 		return nil, trace.NotFound("empty auth servers list")
 	}
-	var errors []error
-
-	// iterate over the addresses in random order
-	for len(authServers) > 0 {
```

Comment:
Ditto here, any implications in dropping this loop?

Reply:
This change (and the one in `proxySubsys`) are more controversial; we discussed this in the scaling meeting and the consensus was that client (this will affect `tsh` and Connect) connection failures with a properly configured auth load balancer (the HA setup we recommend and support) will be few and far between, and any retry logic should be implemented in the client, if it turns out to be necessary. Another point in favor of only triggering one connection attempt per client connection is to prevent network amplification impact from a malicious or misconfigured client - or at least reduce it, as `net.Dial` on a hostname might attempt to connect to all the resolved addressed in sequence, but we don't specialcase that anywhere in the codebase.